node-proxy/lib/proxy/ProxyServer.js

```diff
@@ -453,10 +453,21 @@ function finish_websocket(upg_reqhost, proxy_server, ws) {
   proxy_server.debug()  &&  Logger.debug('Sending a websocket request to %s',
                                          util.inspect(ws_endpoint));
 
+  var zheaders = { 'headers': {}};
+
+  zheaders.headers['user-agent'] = upgrade_req.headers['user-agent'];
+
+  /*  Set X-Forwarded-For HTTP extension header.  */
+  var xff = upgrade_req.connection.remoteAddress  ||
+            upgrade_req.socket.remoteAddress;
+  zheaders.headers['X-Forwarded-For'] = xff;
+
+  /*  Set X-Client-IP HTTP extension header.      */
+  zheaders.headers['X-Client-IP'] = xff;
+
```

Comment:
not particularly, but i'd second your belief that case is irrelevant, so it's basically aesthetics.